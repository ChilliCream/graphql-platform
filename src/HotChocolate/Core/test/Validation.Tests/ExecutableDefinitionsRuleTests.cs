using HotChocolate.Language;
using Xunit;

namespace HotChocolate.Validation
{
    public class ExecutableDefinitionsRuleTests
        : ValidationTestBase
    {
        public ExecutableDefinitionsRuleTests()
            : base(new ExecutableDefinitionsRule())
        {
        }

        [Fact]
        public void QueryWithTypeSystemDefinitions()
        {
            // arrange
            Schema schema = ValidationUtils.CreateSchema();
            DocumentNode query = Utf8GraphQLParser.Parse(@"
                query getDogName {
                    dog {
                        name
                        color
                    }
                }

                extend type Dog {
                    color: String
                }
            ");

            // act
            QueryValidationResult result = Rule.Validate(schema, query);

            // assert
            Assert.True(result.HasErrors);
            Assert.Collection(result.Errors,
                t => Assert.Equal(
                    "A document containing TypeSystemDefinition " +
                    "is invalid for execution.", t.Message));
        }

        [Fact]
        public void QueryWithoutTypeSystemDefinitions()
        {
            // arrange
            Schema schema = ValidationUtils.CreateSchema();
            DocumentNode query = Utf8GraphQLParser.Parse(@"
                query getDogName {
                    dog {
                        name
                        color
                    }
                }
            ");

            // act
            QueryValidationResult result = Rule.Validate(schema, query);

            // assert
            Assert.False(result.HasErrors);
            Assert.Empty(result.Errors);
        }
    }
}
