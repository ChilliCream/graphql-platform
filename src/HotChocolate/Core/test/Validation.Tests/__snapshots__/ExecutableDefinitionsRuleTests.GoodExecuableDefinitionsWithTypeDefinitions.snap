[
  {
    "Message": "A document containing TypeSystemDefinition is invalid for execution.",
    "Code": null,
    "Path": null,
    "Locations": [
      {
        "Line": 7,
        "Column": 17
      }
    ],
    "Extensions": {
      "specifiedBy": "http://spec.graphql.org/June2018/#sec-Executable-Definitions"
    },
    "Exception": null,
    "SyntaxNode": {
      "Kind": "ObjectTypeDefinition",
      "Description": null,
      "Interfaces": [],
      "Fields": [
        {
          "Kind": "FieldDefinition",
          "Description": null,
          "Arguments": [],
          "Type": {
            "Kind": "NamedType",
            "Location": {
              "Start": 177,
              "End": 201,
              "Line": 8,
              "Column": 27
            },
            "Name": {
              "Kind": "Name",
              "Location": {
                "Start": 177,
                "End": 201,
                "Line": 8,
                "Column": 27
              },
              "Value": "String"
            }
          },
          "Location": {
            "Start": 171,
            "End": 201,
            "Line": 8,
            "Column": 21
          },
          "Name": {
            "Kind": "Name",
            "Location": {
              "Start": 171,
              "End": 176,
              "Line": 8,
              "Column": 21
            },
            "Value": "name"
          },
          "Directives": []
        }
      ],
      "Location": {
        "Start": 140,
        "End": 224,
        "Line": 7,
        "Column": 17
      },
      "Name": {
        "Kind": "Name",
        "Location": {
          "Start": 145,
          "End": 150,
          "Line": 7,
          "Column": 22
        },
        "Value": "Cow"
      },
      "Directives": []
    }
  }
]
